cmd/annotations-checker/validate-annotations.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-github/v32/github" // GitHub API client library
+	// "github.com/ministryofjustice/cloud-platform-environments/pkg/namespace"
+	"golang.org/x/oauth2" // API requests authentication
+	// "google.golang.org/genproto/googleapis/api/annotations"
+	"gopkg.in/yaml.v2"
+)
+
+// Annotations structs to define the structure of annotations in the YAML file
+type Annotations struct {
+	Kind           string `yaml:"kind"`
+	SourceCodeRepo string `yaml:"cloud-platform.justice.gov.uk/source-code"`
+	TeamName       string `yaml:"cloud-platform.justice.gov.uk/team-name"`
+}
+
+// hold the structure for the YAML file
+type KubernetesConfig struct {
+	Metadata struct {
+		Annotations Annotations `yaml:"annotations"`
+	} `yaml:"metadata"`
+}
+
+// gitHub client
+func NewGitHubClient(token string) *github.Client {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	return client
+}
+
+// validate team name and source code repo
+func ValidateAnnotations(client *github.Client, org string) (string, error) {
+	// get namespace directory
+	namespacesDir := "/path/to/namespaces"
```

Comment:
update this, you need to find a way of passing the files that have been changed/ added and the namespace(s) that are mentioned from the PR to the code